llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -2990,13 +2994,8 @@ SDValue SITargetLowering::LowerFormalArguments(
   }
 
   // Start adding system SGPRs.
-  if (IsEntryFunc) {
+  if (IsEntryFunc)
     allocateSystemSGPRs(CCInfo, MF, *Info, CallConv, IsGraphics);
-  } else {
-    CCInfo.AllocateReg(Info->getScratchRSrcReg());
-    if (!IsGraphics)
```

Comment:
This check limits the call to `allocateSpecialInputSGPRs` from not calling for amdgpu_gfx functions. Are the ABI registers allocated by this function the same for both amdgpu device and gfx functions?

Reply:
No, amdgpu_gfx isn't supposed to get the same inputs. I'm not sure how the workgroup IDs are supposed to be treated for amdgpu_gfx

Reply:
This is actually a bug fix to better match the DAG behavior

Reply:
There is an ongoing patch in the downstream where I enabled architected SGPRs for all callable functions. A patch will be posted in the upstream for gfx12 soon. With that, the call to `allocateSpecialInputSGPRs` needs to be enabled for gfx functions too. We need more clarity on how these special allocations should be treated for gfx functions.

Reply:
amdgpu_gfx isn't going to have any of the HSA inputs which are passed here. It should only maybe have the workgroup / workitem IDs, but they seem to not